### PR TITLE
Ensure file, include and exclude specs used are strings 

### LINKED ISSUE #38164: TS Server fatal error:  path.replace is not a function
We have written the needed data into your clipboard because it was too large to send. Please paste.

**TypeScript Version:** 3.8.3
**Steps to reproduce crash**
1.
2.
3.
**TS Server Error Stack**
```
TypeError: path.replace is not a function
    at normalizeSlashes (tsserver.js:6191:21)
    at Object.combinePaths (tsserver.js:6252:28)
    at getWildcardDirectories (tsserver.js:28441:48)
    at matchFileNames (tsserver.js:28299:35)
    at getFileNames (tsserver.js:27850:26)
    at parseJsonConfigFileContentWorker (tsserver.js:27782:18)
    at Object.parseJsonSourceFileConfigFileContent (tsserver.js:27740:16)
    at ProjectService.loadConfiguredProject (tsserver.js:142263:44)
    at ProjectService.createAndLoadConfiguredProject (tsserver.js:142240:22)
    at ProjectService.createLoadAndUpdateConfiguredProject (tsserver.js:142245:36)
    at ProjectService.assignProjectToOpenedScriptInfo (tsserver.js:143101:44)
    at tsserver.js:143396:107
    at Object.mapDefined (tsserver.js:562:30)
    at ProjectService.applyChangesInOpenFiles (tsserver.js:143396:48)
    at Session.handlers.ts.createMapFromTemplate._a.<computed> (tsserver.js:144260:46)
    at tsserver.js:146003:88
    at IOSession.Session.executeWithRequestId (tsserver.js:145994:28)
    at IOSession.Session.executeCommand (tsserver.js:146003:33)
    at IOSession.Session.onMessage (tsserver.js:146027:35)
    at Interface.<anonymous> (tsserver.js:147342:27)
    at Interface.emit (events.js:203:13)
    at Interface._onLine (readline.js:316:10)
    at Interface._normalWrite (readline.js:461:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:203:13)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:276:11)
    at Socket.Readable.push (_stream_readable.js:210:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)
```

### PR DESCRIPTION
**Pull request by sheetalkamat**
Fixes #38164, #39856

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@typescript-bot test this
@typescript-bot user test this
@typescript-bot run dt

**Comment by DanielRosenwasser** [comment]
@typescript cherry-pick this to release-4.0 and LKG

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -2372,28 +2367,33 @@ namespace ts {
...
+        function getSpecsFromRaw(prop: "files" | "include" | "exclude"): PropOfRaw<string> {
+            return getPropFromRaw(prop, isString, "string");
+        }
+
+        function getPropFromRaw<T>(prop: "files" | "include" | "exclude" | "references", validateElement: (value: unknown) => boolean, elementTypeName: string): PropOfRaw<T> {
+            if (hasProperty(raw, prop) && !isNullOrUndefined(raw[prop])) {
+                if (isArray(raw[prop])) {
+                    const result = raw[prop];
+                    if (!sourceFile && !every(result, validateElement)) {
+                        errors.push(createCompilerDiagnostic(Diagnostics.Compiler_option_0_requires_a_value_of_type_1, prop, elementTypeName));
```
Comment (sandersn) [inline src/compiler/commandLineParser.ts:2387]: should we return something besides `result` in this error case?

Comment (sheetalkamat) [reply src/compiler/commandLineParser.ts:2387]: It’s not needed at moment.. if needed we can add later

**Comment by sheetalkamat** [comment]
@typescript cherry-pick this to release-4.0 and LKG

**Comment by sheetalkamat** [comment]
@typescript cherry-pick this to release-4.0

**Comment by sheetalkamat** [comment]
@typescript-bot  cherry-pick this to release-4.0 and LKG

**Comment by j-oliveras** [comment]
@sheetalkamat #39856 is still open but should be fixed by this PR.

**Comment by sheetalkamat** [comment]
Thanks closed it

**Pull request merged**